### ISSUE TITLE
bug: Appium 2.0 - not starting tests onto Android 13

### ISSUE THREAD

**Issue by mtz29** [appium/appium#18717]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Appium gets stuck on (loop until timeout): 

```
[debug] [ADB] Waiting for emulator startup. Intermediate error: Error executing adbExec. Original error: 'Command '/usr/lib/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell cmd reboot_readiness check-subsystems-state --list-blocking' exited with code 1'; Command output: Subsystem state: true
```

If emulator is already running, second run will work.

### Expected Behavior

It should install apps and start up test cases

### Minimal Reproducible Example

```python

import unittest
from unittest import TestCase
import pytest
import os
from appium import webdriver
from appium.options.android import UiAutomator2Options
import requests


class BaseTest(unittest.TestCase):

    # setUpClass method will only be ran once, and not every suite/test
    @classmethod
    def setUpClass(cls) -> None:
        # reset wiremock scenarios
        resp = requests.post("http://127.0.0.1:8080/__admin/scenarios/reset")
        assert resp.status_code == 200, "Cannot reset wiremock scenarios. Check wiremock instance"

        cls.capabilities = dict(
            platformName='Android',
            platformVersion='13',
            automationName='uiautomator2',
            # deviceName='OnePlus LE2123',
            deviceName='Emulator',
            appPackage='<EDITED>',
            appActivity='.SplashActivity',
            language='pl',
            locale='PL',
            ignoreHiddenApiPolicyError=True,
            autoLaunch=False,
            autoGrantPermissions=True,
            app="<EDITED>",
            udid= os.getenv("EMULATOR_ID", "emulator-5554"),
            # avd="Pixel_6_Pro_API_33", # not working
            avd="Pixel_API_33", # not working
            # avd="Pixel_2_API_30", # this works !!!
            
        )

        options = UiAutomator2Options().load_capabilities(cls.capabilities)
        appium_server_url = 'http://localhost:4723/wd/hub'
        cls.driver = webdriver.Remote(command_executor=appium_server_url, options=options)

        # reverse port forwarding
        os.system("adb -s {} reverse tcp:8080 tcp:8080".format(cls.capabilities["udid"]))

        # clear data
        cls.driver.execute_script('mobile: shell', {'command': 'pm clear', 'args':' {}'.format(cls.capabilities["appPackage"])})
        

    @classmethod
    def startApp(cls, activity=".SplashActivity") -> None:
        # cls.driver.start_activity(app_package=cls.capabilities["appPackage"], app_activity=cls.capabilities["appActivity"])
        cls.driver.start_activity(app_package=cls.capabilities["appPackage"], app_activity=activity)


    @classmethod
    def tearDownClass(cls) -> None:
        # Do cleanup, close the driver, ...
        cls.driver.quit()
```

```
import requests
from .base import BaseTest
import unittest
from unittest import TestCase
import logging
import os
from selenium.webdriver.common.by import By
from appium.webdriver.common.mobileby import MobileBy
from appium.webdriver.common.appiumby import AppiumBy
from appium.webdriver.webelement import WebElement

class MainActivityTest(BaseTest):
    @classmethod
    def setUpClass(cls) -> None:
       super(MainActivityTest, cls).setUpClass()

    def start_activity(self):
        self.startApp(activity=".MainActivity")

    def setUp(self) -> None:
        # This is executed before every test
        resp = requests.post("http://127.0.0.1:8080/__admin/scenarios/reset")
        assert resp.status_code == 200, "Cannot reset wiremock scenarios. Check wiremock instance"

    def testOrder(self):
        resp = requests.put("http://127.0.0.1:8080/__admin/scenarios/refresh-token/state", json={"state": "valid-token"})
        assert resp.status_code == 200, "Cannot set refresh-token wiremock scenario state. Check wiremock instance"        
        
        self.start_activity()


    
    def tearDown(self) -> None:
        # This is executed after every test
        self.driver.terminate_app(self.capabilities["appPackage"])

if __name__ == '__main__':
    unittest.main()
```

### Environment

- Operating system:  Kubuntu 23.04 (`6.2.0-20-generic #20-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  6 07:48:48 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`)
- If running via `appium` CLI...
  - Appium CLI version (output of `appium --version`): 
  - Appium package version: 2.0.0-beta.70 & 2.0.0-beta.71
  - Node.js version (output of `node --version`): v16.15.1
  - `npm` version (output of `npm --version`): 8.4.0
  - Last `appium` version which did _not_ exhibit the problem: 1.X
- Appium driver(s) and their version(s):  appium-uiautomator2-driver:2.25.1
- Appium plugin(s) and their version(s): none
- Platform and version under test: Android 13 (SDK 33)
- Real device or emulator/simulator: emulator/avd

**Comment by mykola-mokhnach** [comment]
Looks like it is an adb bug. the output seems correct, but the exit code does not make much sense to me. Will add a workaround to it.

**Comment by mykola-mokhnach** [comment]
Could you attach the full server log?

**Comment by mtz29** [comment]
So it looks like the issue was introduced last week: https://github.com/appium/appium-adb/commit/c98fbcd4a713296ab5e3a99f75f5cfa2233157cb

Was that ever tested against API 33? What adb version?

**Comment by mykola-mokhnach** [comment]
Perhaps you are looking for https://cs.android.com/android/platform/superproject/+/master:packages/modules/Scheduling/service/java/com/android/server/scheduling/RebootReadinessShellCommand.java

The patch has been published in uia2 driver 2.25.2
